gpt_engineer/core/code.py

```diff
@@ -0,0 +1,13 @@
+from typing import TypeVar, Dict
+
+class Code(dict):
+
+    def __setitem__(self, key, value):
+        if not isinstance(key, str):
+            raise TypeError("Keys must be strings")
+        if not isinstance(value, str):
+            raise TypeError("Values must be integers")
+        super().__setitem__(key, value)
```

Comment:
Nitpick:

Does this work?:)

```suggestion
        super()[key] = value
```

Reply:
No it doesn't! Didn't take time to dig into why